ecleankernel/layout/blspec.py

```diff
@@ -149,6 +146,23 @@ def find_kernels(self,
         for mkv, fobjs in module_dict.items():
             if any(mkv == k.real_kv for k in kernels.values()):
                 continue
-            kernels.setdefault(mkv, Kernel(mkv)).all_files.extend(fobjs)
+            # this mkv does not have a corresponding kernel
+            # with same real_kv in kernels
+            for layout in ("bls", "uki"):
```

Comment:
Let's not hardcode list of possible layouts, just iterate over everything in `kernels` dict.

Reply:
Right, good point. This is definitely a lot simpler, idk why I wrote this so difficultly yesterday.